File: p35.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if(target <= nums[0]){
            return 0;
        }
        for(int i {1}; i < nums.size(); ++i) {
            if(nums[i] >= target && nums[i - 1] < target)
                return i;
        }
        return nums.size();
    }
};

File: p35.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        nums.insert(0, float('inf'))
        left, right = 0, len(nums)
        mid = (left+right)//2

        while left != mid:
            if nums[mid] < target:
                left = mid
            else:
                right = mid
            mid = (left+right)//2

        return mid
